ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala

```diff
@@ -100,10 +103,28 @@ object LedgerClient {
     for {
       ledgerId <- new LedgerIdentityClient(
         stub(LedgerIdentityServiceGrpc.stub(channel), config.token)
-      )
-        .satisfies(config.ledgerIdRequirement)
+      ).satisfies(config.ledgerIdRequirement)
```

Comment:
how do you set the ledger id in the JSON API on startup for this to work out? Currently users do not have to specify one and forcing this is both clunky and backwards compat breaking.

Reply:
actually I think I’m even more confused. You are not actually skipping the call to the ledger identity service so how does this work without a ledger id?

Reply:
We try to verify the ledger id first (Which so far my understanding is should also work without token) and if that fails due to an auth error we just pick up the ledger id the user specified.

I still think about better ways to solve this without invasive changes.